packages/protocol/contracts/common/ProxyCloneFactory.sol

```diff
@@ -0,0 +1,56 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.5.13;
+
+import "openzeppelin-solidity/contracts/ownership/Ownable.sol";
+
+import "./InitializableProxy.sol";
+
+contract ProxyCloneFactory is Ownable {
+  address public proxyAddress;
+  event ProxyCreated(InitializableProxy proxy);
+
+  function() external payable {}
+
+  /**
+   * @notice Sets the address of the implementation to clone.
+   * @param The address of the implementation to clone.
+   */
+  function setProxyAddress(address _proxyAddress) external onlyOwner {
+    proxyAddress = _proxyAddress;
+  }
+
+  // TODO: Upgrade solc version and import from open-zeppelin instead.
+  /**
+   * @notice Creates an EIP-1167 style clone of the specified `_proxyAddress`.
+   * @param The address of the implementation to clone.
+   * @return The address of the clone.
+   * @dev Copied from open-zeppelin.
+   */
+  function clone(address _proxyAddress) internal returns (address instance) {
```

Comment:
any reason to make this parameterized by `_proxyAddress` instead of closed over the storage `proxyAddress`? I assume there is no gas difference due to the compiler

Reply:
I wanted to deviate from OZ code as little as possible, even for something this trivial. This function is lifted directly from their libraries, which we cannot use because our solc version is too old. I do not feel strongly